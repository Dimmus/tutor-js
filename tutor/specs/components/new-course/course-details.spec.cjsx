{React, pause} = require '../helpers/component-testing'

CourseDetails = require '../../../src/components/new-course/course-details'
{NewCourseStore, NewCourseActions} = require '../../../src/flux/new-course'
{CourseActions, CourseStore} = require '../../../src/flux/course'

COURSE_ID = '1'
COURSE = require '../../../api/user/courses/1.json'


describe 'CreateCourse: entering details', ->

  beforeEach ->
    CourseActions.loaded(COURSE, COURSE_ID)

  it 'sets default field values', ->
    NewCourseActions.set(cloned_from_id: COURSE_ID, name: 'Test but Verify')
    wrapper = shallow(<CourseDetails />)
    expect(
      wrapper.find("FormControl[type=\"number\"][value=1]")
    ).to.have.length(1)
    expect(
      wrapper.find("FormControl[type=\"text\"][value=\"Test but Verify\"]")
    ).to.have.length(1)
    undefined

  it 'updates flux values when edited', ->
    wrapper = mount(<CourseDetails />)
    wrapper.find('.course-details-name .form-control')
      .simulate('change', target: value: 'My Course')
    wrapper.find('.course-details-sections .form-control')
      .simulate('change', target: value: 12)
    expect(NewCourseStore.get('name')).to.equal('My Course')
    expect(NewCourseStore.get('num_sections')).to.equal(12)
    undefined
