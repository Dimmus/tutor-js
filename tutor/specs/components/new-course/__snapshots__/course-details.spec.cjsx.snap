exports[`CreateCourse: entering details matches snapshot 1`] = `
<form
  className="">
  <div
    className="course-details-name form-group">
    <input
      autoFocus={true}
      className="form-control"
      defaultValue="My Course"
      id={undefined}
      onChange={[Function]}
      placeholder="Choose a name for your course"
      type="text" />
  </div>
  <div
    className="course-details-sections form-group">
    <span
      className="input-group">
      <span
        className="input-group-addon">
        Number of sections
      </span>
      <input
        className="form-control"
        id={undefined}
        max={12}
        min="1"
        onChange={[Function]}
        type="number"
        value={12} />
    </span>
    <p
      className="course-details-sections-note">
      <small>
        <i>
          (You can add or remove sections later)
        </i>
      </small>
    </p>
    <p
      className="explain copy-explain">
      Any questions you previously excluded in the Question Library will remain excluded in your new course.
    </p>
  </div>
</form>
`;
